pkg/processor/build/builder.go

```diff
@@ -1863,3 +1868,27 @@ func (b *Builder) resolveFunctionHealthCheckInterval() (time.Duration, error) {
 	}
 	return healthCheckInterval, nil
 }
+
+// resolveNodeSelector resolves builder NodeSelector from function, project and platform NodeSelectors,
+// where function values take precedence over project values, and project values take precedence over platform values
+func (b *Builder) resolveNodeSelector(ctx context.Context) (map[string]string, error) {
+	var builderNodeSelector map[string]string
+	project, err := b.platform.GetFunctionProject(ctx, &b.options.FunctionConfig)
+	if err != nil {
+		return nil, errors.Wrap(err, "Failed to get project for the function")
+	}
+	if b.platform.GetConfig().Kube.IgnorePlatformIfProjectNodeSelectors {
+		builderNodeSelector = common.MergeNodeSelector(b.options.FunctionConfig.Spec.NodeSelector,
+			project.GetConfig().Spec.DefaultFunctionNodeSelector,
+			nil)
+	} else {
+		builderNodeSelector = common.MergeNodeSelector(b.options.FunctionConfig.Spec.NodeSelector,
+			project.GetConfig().Spec.DefaultFunctionNodeSelector,
+			b.platform.GetConfig().Kube.DefaultFunctionNodeSelector)
+	}
```

Comment:
I would just add a comment for each condition to make it more human readable (no need for logs):
"Enriching from platform only"
"Enriching from both platform and project"